core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -74,6 +80,17 @@ public void unbind(ServiceReference<ManagementContext> managementContext) throws
 
     public class CatalogPopulator extends BundleTracker<Iterable<? extends CatalogItem<?, ?>>> {
 
+        private final String BROOKLYN_PACKAGE_REGEX = "org.apache.brooklyn.*";
+
+        private final ConfigKey<List<String>> WHITELIST =
+            ConfigKeys.newConfigKey(new TypeToken<List<String>>(){}, "brooklyn.catalog.osgi.application.whitelist",
+                "CSV Whitelist of patterns to match bundle symbolic id if bundle is to be permitted to add " +
+                    "applications (templates) to the catalog", ImmutableList.of(BROOKLYN_PACKAGE_REGEX));
+        private final ConfigKey<List<String>> BLACKLIST =
+            ConfigKeys.newConfigKey(new TypeToken<List<String>>(){}, "brooklyn.catalog.osgi.application.blacklist",
+                "CSV Blacklist of patterns to match bundle symbolic id to prevent selected whitelisted bundles " +
+                    "adding applications (templates) to the catalog", ImmutableList.<String>of());
+
```

Comment:
Since this is a configuration set at dist build time do you think it will be better if set in [`org.apache.brooklyn.osgilauncher.cfg`](https://github.com/apache/brooklyn-server/blob/master/karaf/init/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L33)? 

`brooklyn.properties` is reserved for users, so they will have to add this key themselves.